index.js

```diff
@@ -85,9 +91,17 @@ module.exports.lintText = (str, opts) => {
 };
 
 module.exports.lintFiles = (patterns, opts) => {
+	const isEmptyPatterns = patterns.length === 0;
+
+	opts = opts || {};
+	if (!isEmptyPatterns && !opts.cwd) {
+		// Use file path rather process.cwd for analysing files
+		const cwd = path.dirname(arrify(patterns)[0]);
```

Comment:
Using the first entry in the array is arbitrary and it would be inconsistent as the behavior would change depending on the order of the input passed by the user.

In addition is will not work because `patterns` contains globs not file paths. So `path.dirname` will fail.